### PR TITLE
AL-579: Updating GLS processing and fixed tests for multi-integrations.

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Resolves [AL-579](https://jira.stsci.edu/browse/AL-579)

Closes #

This PR addresses problems computing final computations for multiple integrations.  This has been fixed and tests have been updated.

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [x] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

**Review by dmggh — APPROVED** [review]
A couple purely editorial comments :

In gls_fit.py, an incredibly picky and time-consuming-to-address comment: some but not all of the functions have all of the docs sections "Summary", "Parameters",  and "Return".  For consistency, functions missing any of these sections should have them added.

File: src/stcal/ramp_fitting/gls_fit.py [hunk 1]
```diff
@@ -552,21 +552,18 @@ def gls_fit_single(ramp_data, gain_2d, readnoise_2d, max_num_cr, save_opt):
     if ngroups == 1:
         med_rates = utils.compute_median_rates(ramp_data)
 
+    # We'll propagate error estimates from previous steps to the
+    # current step by using the variance.
+    input_var_sect = input_var_sect ** 2
+
```
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply src/stcal/ramp_fitting/gls_fit.py:558]: These variable names are changed in another as of yet to be open PR that changes many variables in this function to make more sense.

File: src/stcal/ramp_fitting/gls_fit.py [hunk 2]
```diff
@@ -552,21 +552,18 @@ def gls_fit_single(ramp_data, gain_2d, readnoise_2d, max_num_cr, save_opt):
...
-        # We'll propagate error estimates from previous steps to the
-        # current step by using the variance.
-        # TODO Should this really be done every loop?
-        input_var_sect = input_var_sect ** 2
-
-        # Convert the data section from DN to electrons.
-        # TODO Should this really be done every loop?
-        data_sect *= gain_2d
-        if save_opt:
             first_group[:, :] = data_sect[num_int, 0, :, :].copy()
```
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply src/stcal/ramp_fitting/gls_fit.py:567]: I removed the line at 566.

**Review by hbushouse — APPROVED** [review]

**Comment by hbushouse** [comment]
CI failure for JWST pipeline tests is unrelated (expected differences due to recent updates to NIRSpec reference files in CRDS).

**Pull request merged**